src/database_observer.py

```diff
@@ -58,15 +61,32 @@ def _change_config(self, charm_state: CharmState) -> None:
         Args:
             charm_state: Instance of CharmState
         """
+        if self._charm.get_main_unit() is None and self._charm.unit.is_leader():
```

Comment:
As far as I see, the method database_observer._change_config is the same as the charm.change_config. Why not calling the charm method instead?  (maybe add that method to the base class of the charm, so no casting is needed)

Reply:
I believe this will be better handled/discussed in ISD-1820. What do you think?

Reply:
But...this is related to the last commit that I pushed regarding is_main check. I'll wait for the integration tests to run and see the behavior but I think it would be  easy to do your suggestion in all observers and just call the charm.change_config method.

Reply:
Proposal: Merge this one (after reviews) and then work on the reconcile/restarts in the ISD-1820.

Reply:
Reconcile: https://github.com/canonical/synapse-operator/pull/378